subprojects/build-init/src/main/java/org/gradle/buildinit/tasks/InitBuild.java

```diff
@@ -183,44 +190,61 @@ public void setupProjectLayout() {
         UserInputHandler inputHandler = getServices().get(UserInputHandler.class);
         ProjectLayoutSetupRegistry projectLayoutRegistry = getProjectLayoutRegistry();
 
-        BuildInitializer initDescriptor = null;
-        if (isNullOrEmpty(type)) {
-            BuildConverter converter = projectLayoutRegistry.getBuildConverter();
-            if (converter.canApplyToCurrentDirectory(projectDir)) {
-                if (inputHandler.askYesNoQuestion("Found a " + converter.getSourceBuildDescription() + " build. Generate a Gradle build from this?", true)) {
-                    initDescriptor = converter;
-                }
-            }
-            if (initDescriptor == null) {
-                ComponentType componentType = inputHandler.selectOption("Select type of project to generate", projectLayoutRegistry.getComponentTypes(), projectLayoutRegistry.getDefault().getComponentType());
-                List<Language> languages = projectLayoutRegistry.getLanguagesFor(componentType);
-                if (languages.size() == 1) {
-                    initDescriptor = projectLayoutRegistry.get(componentType, languages.get(0));
-                } else {
-                    if (!languages.contains(Language.JAVA)) {
-                        // Not yet implemented
-                        throw new UnsupportedOperationException();
-                    }
-                    Language language = inputHandler.selectOption("Select implementation language", languages, Language.JAVA);
-                    initDescriptor = projectLayoutRegistry.get(componentType, language);
-                }
-            }
-        } else {
-            initDescriptor = projectLayoutRegistry.get(type);
+        BuildInitializer initDescriptor = getBuildInitializer(inputHandler, projectLayoutRegistry);
+
+        ModularizationOption modularizationOption = getModularizationOption(inputHandler, initDescriptor);
+
+        BuildInitDsl dsl = getBuildInitDsl(inputHandler, initDescriptor);
+
+        boolean useIncubatingAPIs = shouldUseIncubatingAPIs(inputHandler);
+
+        BuildInitTestFramework testFramework = getBuildInitTestFramework(inputHandler, initDescriptor, modularizationOption);
+
+        String projectName = getProjectName(inputHandler, initDescriptor);
+
+        String packageName = getPackageName(inputHandler, initDescriptor, projectName);
+
+        java.util.Optional<JavaLanguageVersion> toolChainVersion = getJavaLanguageVersion(inputHandler, initDescriptor.getLanguage());
```

Comment:
❗ I think this functionality should initially only be added if the "Generate build using new APIs and behavior (some features may change in the next minor release)?" is selected.  This can be changed later, but people might be relying on the old behavior.